blocks/pipeline.py

```diff
@@ -148,3 +155,45 @@ def get_configs(self) -> List[BaseConfig]:
                 ]
             )
         }
+
+    def get_datasource_configs(self, source_type: str) -> Dict[str, BaseConfig]:
+        entire_pipeline = self.children(source_type)
+        return {
+            block.id: vars(block.dataloader.preprocessing_configs[0])
+            for block in flatten(entire_pipeline)
+            if isinstance(block, DataSource)
+        }
+
+    def get_datasource_types(self) -> List[str]:
```

Comment:
can we add a comment to this function that describes what it does and why? definitely not super straightforward to understand where should we call this